### PR TITLE
UIOR-374 update condition for reset instanceId

### PR DESCRIPTION
**Pull request by Feboll**
## Purpose

To allow a user to remove extra product IDs that were imported from an Inventory Instance, without losing the connection to the Instance
https://issues.folio.org/browse/UIOR-374

## Approach

Update condition for reset instanceId

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/components/POLine/Item/util.js [hunk 1]
```diff
@@ -28,10 +28,13 @@ export const checkInstanceIdField = (formValues, inventoryData) => {
...
-  const isEqualProductIds = inventoryData.productIds.every(el => {
-    const productId = find(get(formValues, 'details.productIds', []), { 'productId': el.productId });
 
-    return productId ? isEqual(productId, el) : false;
+  const formProductIds = get(formValues, 'details.productIds', []);
+
+  const isEqualProductIds = formProductIds.every(item => {
+    const productId = find(inventoryData.productIds, { 'productId': item.productId });
+
+    return productId ? isEqual(productId, item) : false;
```
Comment (aliaksei-chumakou) [inline src/components/POLine/Item/util.js]: does it work if user adds product?

Comment (Feboll) [reply src/components/POLine/Item/util.js]: When we add new productId to list we remove the Instance UUID. Also we remove the Instance UUID if change some product.

Comment (Feboll) [reply src/components/POLine/Item/util.js]: We still attached to instance uuid only if removed.

Comment (aliaksei-chumakou) [reply src/components/POLine/Item/util.js]: but earlier it worked - instance uuid was attached if user adds product id and type, as I can see

Comment (Feboll) [reply src/components/POLine/Item/util.js]: as I understand it, we are changing this behavior. Look in the ticket please https://issues.folio.org/browse/UIOR-374

Comment (aliaksei-chumakou) [reply src/components/POLine/Item/util.js]: I looked into the ticket and found there info only about `Allow deletion`. I think the rest behavior should stay the same

Comment (aliaksei-chumakou) [reply src/components/POLine/Item/util.js]: yes, scenario 2 contains that changed condition

**Pull request merged**